newIDE/app/src/UI/Dialog/index.js

```diff
@@ -40,26 +44,33 @@ const styles = {
 type Props = {|
   open?: boolean,
   title?: React.Node,
-  actions?: React.Node,
-  secondaryActions?: React.Node,
+  actions?: Array<?React.Node>,
+  secondaryActions?: Array<?React.Node>,
 
   /**
    * Callback called when the dialog is asking to be closed
    * (either by Escape key or a click outside, according to preferences).
    * This is the default way of closing a dialog and should almost always be
-   * specified - unless your dialog is an representing an uninteruptible process.
+   * specified - unless your dialog is representing an uninteruptible process.
    *
    * If `onApply` is also specified, this must be interpreted as a "cancelling"
    * of changes.
    */
   onRequestClose?: () => void,
   /**
-   * Is specified, will be called when the dialog is dismissed in a way where changes
+   * If specified, will be called when the dialog is dismissed in a way where changes
    * must be kept.
    * This is not applicable to all dialogs. Some dialogs may have no `onApply` and just a
    * single `onRequestClose`.
    */
   onApply?: ?() => void,
+  /**
+   * If specified, allows user to close the dialog with an action based on
+   * `shouldSubmit` key event. `lastAction` option simulates a click on the
+   * last (often the main) action button given (if this last action is a button
+   * with split menu, it simulates click on the main button).
+   */
+  onSubmit?: 'lastAction' | (() => void),
```

Comment:
We should avoid props that can have different types (here, `string` and `function`)
We should either have `?string` (and if null, decide to not do anything in the logic)
Or we could also decide to pass the function we want triggered when the submit key is called instead of `lastAction` which would allow us in the future to have some custom logic in there, that would be different than when the user clicks on the action?

Reply:
If we change the `onSubmit` to be a function, this would add some duplication when defining a dialog, but at least it would be clear what it does + you wouldn't need the `findAndClickButton` function

Reply:
As a user of the component of the Dialog, I don't wanna spend time thinking about this, because I think that 99% of the time, the onSubmit will consist of simulate a click on the last action (proof is that the possibility to pass a function is not used in all the dialogs I reviewed).
I can instead use a props `clickOnLastActionOnSubmit: boolean = false`

Reply:
I like having to think what all actions should do when I create a dialog as a dev :) but you're right saying this hasn't been used consistently.
Then yes, let's use a boolean then, `submitLastAction: boolean  = false` ? or `canSubmitLastAction` ?